examples/api/main.go

```diff
@@ -31,29 +39,61 @@ func main() {
 
 	defer l.Sync()
 
-	l.Info("listening", zap.String("addr", addr))
+	ctx := telemetry.ContextWithLog(context.Background(), telemetry.NewZapLog(l))
 
-	http.ListenAndServe(addr, httpx.NewAPIRouter(httpx.APIRouterConfig{
-		APIRouters:  newAPIMap(),
-		ServiceName: "birds",
-		Logger:      l,
-	}))
+	svc, err := service.New(ctx, service.Config{
+		Name:         serviceName,
+		BuildVersion: serviceVersion,
+		BuildHash:    "abc111",
+		TracerProviderConfig: telemetry.TracerProviderConfig{
+			ServiceName:    serviceName,
+			ServiceVersion: serviceVersion,
```

Comment:
I think this can be punted to whatever is using this package. Consumers are free to pass whatever data they want as the "version" (hash, semver, etc).

Reply:
Or, its actually probably better to accept both and and have the version and the hash on both the service and telemetry.